test/integration/scheduler_perf/util.go

```diff
@@ -203,21 +206,33 @@ func (*metricsCollector) run(ctx context.Context) {
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
 	for _, metric := range pc.Metrics {
-		dataItem := collectHistogramVec(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
+		metricFamily, err := testutil.GetMetricFamilyFromGatherer(legacyregistry.DefaultGatherer, metric)
+		if err != nil {
+			klog.Error(err)
+			return dataItems
+		}
+		if metric == extentionPointsMetricName {
+			for _, value := range extentionPoints {
+				dataItem := generateData(metricFamily, pc.labels, map[string]string{extentionPointsLabelName: value}, metric)
+				if dataItem != nil {
+					dataItems = append(dataItems, *dataItem)
+				}
+			}
+		} else {
+			dataItem := generateData(metricFamily, pc.labels, nil, metric)
+			if dataItem != nil {
+				dataItems = append(dataItems, *dataItem)
+			}
 		}
 	}
 	return dataItems
 }
 
-func collectHistogramVec(metric string, labels map[string]string) *DataItem {
-	vec, err := testutil.GetHistogramVecFromGatherer(legacyregistry.DefaultGatherer, metric, nil)
-	if err != nil {
-		klog.Error(err)
+func generateData(metricFamily *dto.MetricFamily, labels, lvMap map[string]string, metric string) *DataItem {
```

Comment:
why `generateData`? Could it be `collectHistogramData`?

Reply:
sound better, updated.